### PR TITLE
tests: libvmi.New: no explicit name

### PR DESCRIPTION
**Pull request by dankenigsberg**
In continuation to PR #7850, this PR suggests to further simplify the API of `libvmi` by dropping the compulsory `name` argument, which is never used by us. This makes it simpler to reimplemnt `tests.NewRandomVMIWithNS` with `libvmi`. If we ever need to set the VMI name, we can add a WithName() builder option.

```release-note
NONE
```

### CODE REVIEW

File: tests/utils.go [hunk 1]
```diff
@@ -996,24 +996,15 @@ func NewRandomVMI() *v1.VirtualMachineInstance {
...
-	vmi.Spec.TerminationGracePeriodSeconds = &t
-
 	// To avoid mac address issue in the tests change the pod interface binding to masquerade
 	// https://github.com/kubevirt/kubevirt/issues/1494
-	vmi.Spec.Domain.Devices = v1.Devices{Interfaces: []v1.Interface{{Name: "default",
-		InterfaceBindingMethod: v1.InterfaceBindingMethod{
-			Masquerade: &v1.InterfaceMasquerade{}}}}}
+	vmi := libvmi.New(
+		libvmi.RandName(),
+		libvmi.WithMasqueradeNetworking()...,
```
Comment (EdDev) [inline tests/utils.go:998]: `WithMasqueradeNetworking` is confusing. It includes also cloud-init network config, which I think you should not have here.

Comment (dankenigsberg) [reply tests/utils.go]: I don't understand, but let me try it. Another option is to drop Masquerade altogether, given the fact that https://github.com/kubevirt/kubevirt/issues/1494 is resolved in libvirt. But this should way for another PR and further discussion.

Comment (EdDev) [reply tests/utils.go]: The one you added includes cloud-init configuration of the guest, while the one that existed until now does not.

**Comment by EdDev** [comment]
@dankenigsberg , the logs say:
```
tests/utils.go:998:24: undefined: InterfaceDeviceWithMasqueradeBinding
tests/utils.go:999:22: undefined: kvirtv1 
```

File: tests/vmi_lifecycle_test.go [hunk 2]
```diff
@@ -1306,7 +1306,11 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 				node := nodes.Items[0].Name
 
 				By("Creating a VirtualMachineInstance with different namespace")
-				vmi = tests.NewRandomVMIWithNS(*namespace)
+				vmi = libvmi.New(
+					libvmi.WithResourceMemory("1Mi"),
```
Comment (EdDev) [inline tests/vmi_lifecycle_test.go:1310]: Where is this coming from?
Is it due to the missing arch check? I am fine to use `256Mi` to cover it as well.

Comment (dankenigsberg) [reply tests/vmi_lifecycle_test.go:1310]: This test does not have a running guest; any memory provided to the guest is wasted regardless of the achitecture. Should I dedicate a separate commit to explain this change, as this is *not* pure inlining?

Comment (EdDev) [reply tests/vmi_lifecycle_test.go:1310]: I see the domain is started and therefore running.
I think the memory is already assigned to the qemu process once it is started.

I would just use the minimum that fits all arch (i.e. 256Mi).

Comment (dankenigsberg) [reply tests/vmi_lifecycle_test.go:1310]: > I would just use the minimum that fits all arch (i.e. 256Mi).

I'll do that to get this PR approved, and follow up with a separately motivated PR for memory reduction.

**Review by EdDev — APPROVED** [review]
Thank you!

**Review by enp0s3 — APPROVED** [review]
/approve

**Pull request merged**